packages/sdk/src/evm/contracts/smart-contract.ts

```diff
@@ -316,12 +317,22 @@ export class SmartContract<TContract extends BaseContract = BaseContract>
     return undefined;
   }
 
-  private detectAppURI() {
-    if (
-      detectContractFeature<AppURI>(this.contractWrapper, "AppURI") ||
-      detectContractFeature(this.contractWrapper, "ContractMetadata")
+  private detectApp() {
+    const metadata = new ContractMetadata(
+      this.contractWrapper,
+      CustomContractSchema,
+      this.storage,
+    );
+
+    if (detectContractFeature<IAppURI>(this.contractWrapper, "AppURI")) {
+      return new ContractAppURI(this.contractWrapper, metadata);
+    } else if (
```

Comment:
These can just be in a single if statement with an `||`

Reply:
sounds good

Reply:
didn't work, leaving it